src/components/Transcript.tsx

```diff
@@ -0,0 +1,347 @@
+import { formatRelativeDay } from "../recordings-utils";
+import "@brave/leo/tokens/css/variables.css";
+import Button from "@brave/leo/react/button";
+import Input from "@brave/leo/react/input";
+
+interface MeetingTranscriptProps {
+  transcript: DownloadedTranscript;
+}
+
+interface MeetingTranscriptDisplayProps {
+  transcriptId: string;
+  transcriptUrlBase?: string;
+}
+
+const styles = {
+  row: css`
+    display: flex;
+    padding: var(--leo-spacing-m, 0px);
+    align-items: flex-start;
+    gap: var(--leo-spacing-xl);
+    align-self: stretch;
+    line-height: var(--Line-height-Large, 24px);
+    font-size: var(--Size-Large, 16px);
+    font-family: var(--Family-Default, "Inter Variable"), Inter;
+  `,
+  cell: css``,
+  actor: css`
+    width: 128px;
+    font-style: normal;
+    font-weight: 600;
+    letter-spacing: -0.2px;
+  `,
+  action: css`
+    font-style: italic;
+    color: var(--leo-color-systemfeedback-error-text);
+    flex: 1 0 0;
+    font-weight: 400;
+    letter-spacing: -0.2px;
+  `,
+  message: css`
+    flex: 1 0 0;
+    color: var(--leo-color-text-primary);
+    font-style: normal;
+    font-weight: 400;
+    letter-spacing: -0.2px;
+  `,
+};
+
+const COLORS = [
+  "var(--leo-color-secondary-40)",
+  "var(--leo-color-primary-40)",
+  "var(--leo-color-green-40)",
+  "orange",
+  "#BADA55",
+  "purple",
+];
+
+export const MeetingTranscript = ({ transcript }: MeetingTranscriptProps) => {
+  const participantColorMap = new Map<string, string>();
+  let participantCounter = 0;
+  const { events, startDateTime } = transcript;
+  events.forEach((e) => {
+    const color = participantColorMap.get(e.participant);
+    if (!color) {
+      participantColorMap.set(
+        e.participant,
+        COLORS[participantCounter++ % COLORS.length],
+      );
+    }
+  });
+
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const textRef = createRef<HTMLDivElement>();
+  useEffect(() => {
+    if (!window.CSS?.highlights) {
+      return;
+    }
+    CSS.highlights.clear();
+    const str = searchTerm.trim().toLowerCase();
+    if (!str || !textRef.current) {
+      return;
+    }
+    const ranges: Range[] = [];
+    for (const rowElement of textRef.current.childNodes) {
+      const textNode = rowElement.lastChild?.firstChild;
+      if (!textNode) {
+        continue;
+      }
+      const content = textNode?.textContent?.toLowerCase();
+      if (!content) {
+        continue;
+      }
+      let index = content.indexOf(str);
+      while (index > -1) {
+        const range = new Range();
+        range.setStart(textNode, index);
+        range.setEnd(textNode, index + str.length);
+        ranges.push(range);
+
+        index = content.indexOf(str, index + 1);
+      }
+    }
+    console.log(ranges);
+    const highlight = new Highlight(...ranges);
+    CSS.highlights.set("search-results", highlight);
+  }, [searchTerm]);
+
+  return (
+    <div
+      css={css`
+        margin: 93px auto;
```

Comment:
perhaps we should keep most styling in `styles` above, so the tree is easier to read